### DISCUSSION TITLE
How do you reproduce the response curves?

### DISCUSSION THREAD

**Discussion by Snowcatcat** [facebookexperimental/Robyn#334] [Q&A]
<img width="322" alt="Response" src="https://user-images.githubusercontent.com/99045100/156701963-f86405f3-5ad9-4356-abfa-30d0437d148d.PNG">

What are the differences among:
1. predictedExposure	
2. adstockedMedia 
3. saturatedMedia	
4. saturatedSpendReversed	
5. decompMedia

Let's say we also know the values for gamma, alpha, and theta, how can we reproduce the response curves shown in the one-pager?

Thanks for the help!!!!

**Comment by kyletgoldberg** [comment]
For the latter question - the robyn_reponse() function should make it quite easy to recreate the response curves if you run that in a loop. In terms of those different vars, they're all used in the calculations to create the plots that show up in the one-pagers that are output for each model. Were there specific differences you were having trouble with? Some more context would be helpful. Cheers

**Reply by Snowcatcat** [reply]
Thanks a lot for your reply! Would you mind sharing the syntax on how to use robyn_response for a specific curve? 

I'm wondering what are the definitions for saturated media， saturatedSpend， Reversed decomp Media， and how they are calculated?

 I guess predicted exposure is the transformed impression after adstock and saturation, adstocked media is the transformed impression after adstock, and saturated media is the transformed impression after saturation.

**Reply by kyletgoldberg** [reply]
For using Robyn response for specific curve you can use something like this to calculate spend and response pairs and then use those however you like

```
max_spend <- 100000
spend <- 1000
step <- 1000
response_df <- data.frame()
while(spend <= max_spend){
  response <- robyn_response(
    robyn_object = robyn_object
    , media_metric = 'search_S'
    , metric_value = spend
  )
  new_resp <- c(spend, response$response[1])
  response_df <- rbind(response_df, new_resp)
  spend <- spend + step
}
colnames(response_df) <- c('spend', 'response')
```

In terms of the definitions for the other vars - I'd recommend taking a look at the pareto.R file where they are mostly generated - it should help you to understand better how each of those is generated.
